src/build/build_step.go

```diff
@@ -393,12 +393,17 @@ func moveOutputs(state *core.BuildState, target *core.BuildTarget) ([]string, bo
 		tmpOutput := path.Join(tmpDir, target.GetTmpOutput(output))
 		realOutput := path.Join(outDir, output)
 		if !core.PathExists(tmpOutput) {
-			return nil, true, fmt.Errorf("Rule %s failed to create output %s", target.Label, tmpOutput)
+			return nil, true, fmt.Errorf("rule %s failed to create output %s", target.Label, tmpOutput)
 		}
 		outputChanged, err := moveOutput(state, target, tmpOutput, realOutput)
 		if err != nil {
 			return nil, true, err
 		}
+		tmpStat, tmpErr := os.Stat(tmpOutput)
```

Comment:
can you add a TODO to remove this once the investigation is complete please?

Reply:
Yeah, sure